common/predictive-text/worker/model-compositor.ts

```diff
@@ -12,6 +12,11 @@ class ModelCompositor {
     //        doesn't prevent loading on IE.
     //let whitespaceRemover = /.*[\u0009\u000A\u000D\u0020\u00a0\u1680\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000]/iu;
```

Comment:
Note: this regex does _not_ require the `/u` flag. The `/u` flag just allows for, among other things, curly-braced escapes `\u{1234}`; `\u1234` escapes still work.

See https://mathiasbynens.be/notes/es6-unicode-regex

Reply:
Note: I'll still give this one the stampy without this being addressed; it can be fixed in a different PR

Reply:
I see... so you're saying that because none of these are in the SMP range, we're safe?

Reply:
Correct! From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-unicode-escape):

| | |
|-|-|
| `\uhhhh` | Matches the character with the code hhhh (four hexadecimal digits). |
| `\u{hhhh}` | (only when u flag is set) Matches the character with the Unicode value hhhh (hexadecimal digits). |

Also, I just learned that JavaScript's `\s` handles non-ASCII whitespace too!: 

| | |
|-|-|
| `\s` | Matches a white space character, including space, tab, form feed, line feed. Equivalent to `[ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]`... |


(this is not true of JavaScripts `\b` or `\w`, sadly 😞 )